lib/autoproj/package_manifest.rb

```diff
@@ -181,10 +184,48 @@ class Loader < BaseLoader
 
             MANIFEST_CLASS = PackageManifest
 
-            def initialize(path, manifest)
+            def self.expand_configuration_variable(var, config)
+                prefix = var[0, 1]
+                var = var[1..-1] if prefix == "$"
+
+                if var.start_with?("operating_system_name_")
+                    os = config.get("operating_system", nil)
+                    return "" if os.nil?
+                    os_names, = os
+                    return "" unless os_names.any? do |name|
+                        var == "operating_system_name_#{name}"
+                    end
+                    return "true"
+                end
+
+                if var.start_with?("operating_system_version_")
+                    os = config.get("operating_system", nil)
+                    return "" if os.nil?
+                    _, os_versions = os
+                    return "" unless os_versions.any? do |ver|
+                        var == "operating_system_version_#{ver.gsub(/[.,+-]/, '_')}"
+                    end
+                    return "true"
+                end
+
+                config.get(var)
+            rescue Autoproj::ConfigError
+                ""
```

Comment:
I'm not sure about that one. I would expect the build to fail if a condition is not available. It would avoid people starting to randomly wonder why things don't build ...

Reply:
I can allow the exception to bubble up, the error message seems to be informative enough in that case: `ERROR: undeclared option 'BAR'`. I will repackage the error message in an Autobuild::PackageException, like is done for REXML::ParseException(and fix the use of the uninitialised "file" variable).